src/transaction_simulator.rs

```diff
@@ -267,17 +282,154 @@ impl TransactionSimulator for JunoManager {
     }
 }
 
+async fn block_transaction_to_broadcasted_transaction(
+    juno_manager: &JunoManager,
+    transaction: &Transaction,
+    block_id: BlockId,
+) -> Result<BroadcastedTransaction, ManagerError> {
+    match transaction {
+        Transaction::Invoke(invoke_transaction) => match invoke_transaction {
+            InvokeTransaction::V0(_) => Err(ManagerError::InternalError("V0 invoke".to_string())),
+            InvokeTransaction::V1(tx) => Ok(BroadcastedTransaction::Invoke(
+                BroadcastedInvokeTransaction::V1(BroadcastedInvokeTransactionV1 {
+                    sender_address: tx.sender_address,
+                    calldata: tx.calldata.clone(),
+                    max_fee: tx.max_fee,
+                    signature: tx.signature.clone(),
+                    nonce: tx.nonce,
+                    is_query: false,
+                }),
+            )),
+            InvokeTransaction::V3(tx) => Ok(BroadcastedTransaction::Invoke(
+                BroadcastedInvokeTransaction::V3(BroadcastedInvokeTransactionV3 {
+                    sender_address: tx.sender_address,
+                    calldata: tx.calldata.clone(),
+                    signature: tx.signature.clone(),
+                    nonce: tx.nonce,
+                    resource_bounds: tx.resource_bounds.clone(),
+                    tip: tx.tip,
+                    paymaster_data: tx.paymaster_data.clone(),
+                    account_deployment_data: tx.account_deployment_data.clone(),
+                    nonce_data_availability_mode: tx.nonce_data_availability_mode,
+                    fee_data_availability_mode: tx.fee_data_availability_mode,
+                    is_query: false,
+                }),
+            )),
+        },
+        Transaction::L1Handler(_) => Err(ManagerError::InternalError("L1Handler".to_string())),
```

Comment:
Currently, there's no `BroadcastedTransaction` enum for `L1Handler` transactions. Although it seems like it is supported in the juno client

https://github.com/NethermindEth/juno/blob/84a853b2d71559d37e109d99ca5c7ea0e6136ebc/rpc/handlers.go#L1562
https://github.com/NethermindEth/juno/blob/0820264b6af328d08cf312b4668b7f198e41e213/adapters/sn2core/sn2core.go#L142

Reply:
Since there is none, then let's leave that unsupported